### PR TITLE
Update docs for Swift 3/Moya interface

### PR DESCRIPTION
**Pull request by nanoxd**
Currently a WIP to gauge interest. Updates docs to reflect the new Swift 3 interface as well as Moya's interface.

- [x] Update enum usage
- [x] Match Swift 3 interface changes (e.g. NSURLSession -> URLSession)
- [x] Change usage of `AnyObject` to `Any`
- [x] Reflect interface changes in Moya

### CODE REVIEW

**Comment by BasThomas** [comment]
YES! Thanks for this :)

**Comment by nanoxd** [comment]
I believe that should be it but I'm sure I missed something 😅.

File: docs/Endpoints.md [hunk 1]
```diff
@@ -117,43 +117,41 @@ provide compile-time checking of well-defined network targets. You've already
...
 It's here that you'd do your OAuth signing or whatever. Since you may invoke the
 closure asynchronously, you can use whatever authentication library you like ([example](https://github.com/rheinfabrik/Heimdallr.swift)).
 Instead of modifying the request, you could simply log it, instead.
 
 ```swift
 let requestClosure = { (endpoint: Endpoint<GitHub>, done: MoyaProvider.RequestResultClosure) in
-    // Using the `as!` forced type cast operator is safe here,
-    // as `mutableCopy()` will always return the correct type.
-    let request = endpoint.urlRequest.mutableCopy() as! NSMutableURLRequest
+    var request = endpoint.urlRequest
```
Comment (nanoxd) [inline docs/Endpoints.md]: With the new value types, there's no need to use NSMutableRequest. You instead use a `var` 💃

**Comment by ashfurrow** [comment]
Thanks a lot for contributing to Moya! I've invited you to join the Moya GitHub organization – no pressure to accept! If you'd like more information on what that means, check out our [contributor guidelines](https://github.com/Moya/contributors) and feel free to reach out with any questions.

**Pull request merged**